### PR TITLE
fix(auto-subscribe): auto-subscribe when user comments or likes

### PR DESCRIPTION
**Pull request by mrclay**
This PR has a second commit that removes notify_owner, which could be reverted later.

### CODE REVIEW

File: lib/events.php [hunk 1]
```diff
@@ -1,23 +1,54 @@
...
  *
- * @param string     $event  'create'
- * @param string     $type   'object'
- * @param ElggObject $object The object being created
+ * @param string                    $event  'create'
+ * @param string                    $type   'object'|'annotation'
+ * @param ElggObject|ElggAnnotation $object New comment/discussion reply/like
+ * @return boolean
+ *
+ * TODO Handle group discussion replies once they get migrated to entities
```
Comment (juho-jaakkola) [inline lib/events.php]: Discussion replies are already entities.

**Comment by juho-jaakkola** [comment]
Would it be currently possible to toggle the subscribe/unsubscribe menu item via javascript when someone subscribes using likes?

**Comment by mrclay** [comment]
On the client-side we have a couple problems:

* we can only guess based on HTML what is the "top-level" object that you would be subscribed to
* even if we know, we don't know whether the user has never subscribed or has manually unsubscribed from that object.

Ideally Elgg would allow plugins to inject arbitrary data into the Ajax response so the annotation, create handler could tell the client the answers to both questions.

**Comment by mrclay** [comment]
We could inject into the HTML whether the user has a subscribe block. That would cost a query per entity and solve one problem, but figuring out the correct Subscribe button on the page would still be hard.

**Comment by juho-jaakkola** [comment]
Lists would be easy because they have to ```elgg-object-14163``` class, but yeah for the full_view or anything similar it would be tricky.

**Comment by mrclay** [comment]
Clunky option: Make our own likes button event listener client-side, wait a reasonable amount of time for the real likes action to occur, then fire our our request to get the data we need.

**Comment by beck24** [comment]
I've added autosubscribe on comment back in, it was in the 1.8 version.  Likes I think is a good idea, but we'll have to make the UI make sense

**Pull request closed without merging**